kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/ResourcesUtil.java

```diff
@@ -365,23 +364,20 @@ private static <T extends HasMetadata> boolean isStatusFresh(T resource, Functio
      * valid, null is returned.
      *
      * @param resource resource
-     * @param context context
-     * @param secretEventSourceName event source name used to resolve the secret
      * @param certRef certificate reference
      * @param path path to the certificate reference within the resource
      * @param statusFactory used to generate the condition.
+     * @param secretOpt
      * @return modified resource if the certificate references is invalid, or null otherwise.
      * @param <T> custom resource type
      */
     @Nullable
     public static <T extends CustomResource<?, ?>> T checkCertRef(T resource,
-                                                                  Context<T> context,
-                                                                  String secretEventSourceName,
                                                                   CertificateRef certRef,
                                                                   String path,
-                                                                  StatusFactory<T> statusFactory) {
+                                                                  StatusFactory<T> statusFactory,
+                                                                  Optional<Secret> secretOpt) {
         if (isSecret(certRef)) {
-            Optional<Secret> secretOpt = context.getSecondaryResource(Secret.class, secretEventSourceName);
```

Comment:
It seems add to check the ref, but also pass the Secret. Will it grow more parameters if we support additional kinds in the future? Wouldn't that force the caller to also switch on the reference kind?

Reply:
what I really wanted to do here is resolve everything out of the context once, to avoid different calls to the fabric8 context observing different state during a single reconciliation. Having `Context.getResource` calls buried in ResourceUtil moves in the opposite direction.

I think all the Reconcilers should do something similar, pull everything they can out of the context early and pass it around so we have something stable for checksumming and logic. So I think we should rather be passing in some `ReferentContext` here.

Reply:
hmm, for now I might revert back to having the util do it, but it's going to have to wrap the result up in a record that can pass the Referent back for use in the checksummer. Would rather not get to deep into refactoring dependency resolution in this PR.